## Chi-squared_test by Catherine Ann Celeste Quinlan

## Based on the wikipedia example of workers occupations and areas  A,B,C and D

**Pearsons chi squared test is used here  for Independence for categorical variables .**

In [8]:
import pandas as pd  # importing tools
from scipy.stats import chi2
from scipy.stats import chi2_contingency

table = {'Workers': ['White Collar','Blue Collar','No collar'],   # creating titles for table
         
        'A': [90,30,30],
         
        'B': [60,50,40],
        'C': [104,51,45],        # creating contents of table
        'D': [95,20,35]
        }


df = pd.DataFrame(table, columns = ['Workers', 'A','B','C','D']) # Putting titles and columns together in a dataframe

print (df)  # checking my dataframe 


        Workers   A   B    C   D
0  White Collar  90  60  104  95
1   Blue Collar  30  50   51  20
2     No collar  30  40   45  35


In [9]:
table2 =    [[90,30, 30],         # now keeping the figures seperate  in another table so I can work with them 
			[60,50,40],
            [104,51,45],
             [95,20,35]]




In [10]:
chi2value, p, dof, ex = (chi2_contingency(table2))# It returns a chi2value,a p value,and an estimate of degrees of freedom  before printing expected frequency  table

In [11]:
print('chi2value=%.1f' %chi2value) # printing the values calaculated from the function above
print ('p=%.2f' %p)
print('dof=%.2f' % dof)
print(ex)

chi2value=24.6
p=0.00
dof=6.00
[[ 80.53846154  34.84615385  34.61538462]
 [ 80.53846154  34.84615385  34.61538462]
 [107.38461538  46.46153846  46.15384615]
 [ 80.53846154  34.84615385  34.61538462]]


As required we see a degree of freedom of 6 while our chi squared  value is  rounded to 24.6 .This value is got by not ing the difference right across the table and totalling it 


### Hypothesis testing :







We use the degrees of freedom and a probability of 95%

### Analysis:
    

If our chi squared value is too large we would reject the null hypothesis (of independence).


So how do we find if it is too large ? 

 For this we need to compare this value of 24.6 to what we would get with the degree of freedom of 6 on a Normal distribution for chi squared.

In [14]:
prob = .95
limit = chi2.ppf(prob, dof)# Here we are looking for a value on a normal distribution of chi squared  


In [15]:
print('limit=%.2f' % limit) # This value is printed

limit=12.59


If our value is greater we reject  the null hypothesis of independence, and accept there IS a relationship between districts A,B,c and D  and  workers of white blue and no collar above. 

As we can see our 24.6 is much greater than the 12.59 from a normal chi sqaured distribution, we have to reject the null hypothesis here, and accept that there is a relationship between people's occupation  above and the district  where they live.

[1] https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html

[2]https://machinelearningmastery.com/chi-squared-test-for-machine-learning/

[3]https://www.statisticshowto.com/probability-and-statistics/chi-square/

[4]https://en.wikipedia.org/wiki/Chi-squared_test